### ISSUE TITLE
❗ NOTICE: stepfunctions-tasks: BedrockInvokeModel outputPath no longer passes output to next state

### ISSUE THREAD

**Issue by clareliguori** [aws/aws-cdk#31302]
Please add your +1 👍 to let us know you have encountered this
---

### Overview:
Previously, the BedrockInvokeModel `outputPath` parameter could be used to select a portion of the model output to pass to the next state. This behavior aligns with how the parameter is documented "JSONPath expression to select select a portion of the state output to pass to the next state."
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_stepfunctions_tasks.BedrockInvokeModel.html#outputpath

Now, the outputPath parameter is assumed to be an S3Uri where the model output should be saved. The parameter can no longer be used to select model output and pass it to the next state.

This behavior appears to have changed in https://github.com/aws/aws-cdk/commit/f5dd73bd19f546335531f587f6d7fbde8aff96ce

### Solution:
Revert existing code for encoding the path under S3Uri, and possibly introduce another variable in BedrockTaskProps instead of extending the existing `TaskStateBaseProps`.

### Regression Issue

- [X] Select this option if this issue appears to be a regression.

### Last Known Working CDK Version

2.150.0

### Expected Behavior

In a BedrockInvokeModel task, set outputPath:
```python
# Extract the response from the model
output_path="$.Body.content[0].text",
```

The Step Functions workflow definition is synthesized as:
```
"OutputPath": "$.Body.content[0].text",
```

When I execute the deployed Step Functions state machine, the model text output is passed as the output of the task.
```
"Here is a one sentence summary of ...."
```

### Current Behavior

The outputPath parameter is synthesized as:
```
"Output": {
   "S3Uri": "$.Body.content[0].text"
}
```

When I execute the deployed Step Functions state machine, I get a validation exception:
```
The value for the S3Uri field is not recognized as a valid URI: $.Body.content[0].text
```

### Reproduction Steps

This is my stack:
https://github.com/aws-samples/amazon-bedrock-serverless-prompt-chaining/blob/main/techniques/stacks/model_invocation.py

### Possible Solution

I suggest reverting the change above, so that outputPath works as documented and expected. The `input` and `output` parameters were intended to distinguish between state that should be passed between states (inputPath/outputPath) vs destinations where input and output should be sourced from (input/output).

### CDK CLI Version

2.155.0

### Node.js Version

v20.17.0

### OS

Linux

### Language

Python

### Language Version

Python 3.12.3

**Comment by pahud** [comment]
Hi @clareliguori 

Thanks for the report. We've confirmed this a regression. We'll take actions immediately.

**Comment by shikha372** [comment]
This issue has been fixed now in [latest release (v2.156.0)](https://github.com/aws/aws-cdk/releases/tag/v2.156.0) under  feature flag `useNewS3UriParametersForBedrockInvokeModelTask` which is enabled by default. New properties `s3InputUri` , `s3OutputUri` have been added under [BedrockInvokeModelProps](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/invoke-model.ts#L16)  which can be used to populate S3 URI field dynamically the from previous runtime. To maintain the existing behaviour, you can set this feature flag to `false` but we recommend using the newly added properties.
 
Please let us know if you are still facing the issue.
